src/containers/buttons/GenerateSuiteButton.js

```diff
@@ -0,0 +1,137 @@
+import React, {Component} from "react";
+import {bindActionCreators} from "redux";
+import {connect} from "react-redux";
+import xmlbuilder from "xmlbuilder";
+import {saveAs} from "file-saver";
+
+
+class GenerateSuiteButton extends Component {
+
+  handleSuiteGenerating() {
+    const projectTests = this.props.project[0].tests;
+    const suiteElement = xmlbuilder.create('suite', {encoding: "utf-8"});
+    suiteElement.att('name', this.props.project[0].suite);
+    suiteElement.att('company', this.props.project[0].company);
+    suiteElement.att('domain', this.props.project[0].domain);
+    suiteElement.att('project', this.props.project[0].project);
+    if(Object.values(projectTests).length > 0) {
+      Object.values(projectTests).forEach((testItem) => {
+        const testElement = suiteElement.ele("test", {name: testItem.name.name})
+        this.generateCollectorsGroup(testElement, testItem);
+        this.generateComparatorsGroup(testElement, testItem);
+        this.generateUrls(testElement, testItem);
+      });
+    }
+    const xml = suiteElement.end({pretty: true});
+    var file = new File([xml], "suite.xml", {type: "application/xml;charset=utf-8"});
+    saveAs(file);
+  }
+
+  generateCollectorsGroup(testElement, projectTests) {
+    const collectors = testElement.ele("collect");
+    projectTests.tests.forEach((testItem) => {
+      if(testItem.dropTo === "Modifiers") {
+        const modifierItem = collectors.ele(testItem.tag);
+        this.generateParameters(testItem, modifierItem);
+      } 
+    });
+    collectors.ele("open");
+    projectTests.tests.forEach((testItem) => {
+      if(testItem.dropTo === "Collectors") {
+        const collectorItem = collectors.ele(testItem.tag);
+        this.generateParameters(testItem, collectorItem);
+      }
+    });
+  }
+
+  generateComparatorsGroup(testElement, projectTests) {
+    const comparators = testElement.ele("compare");
+    projectTests.tests.forEach((testItem) => {
+      if(testItem.dropTo === "Comparators") {
+        const collectorItem = comparators.ele(testItem.tag);
+        this.generateParameters(testItem, collectorItem);
+        this.generateFilters(testItem, collectorItem);
+      }
+    });
+  }
+
+  generateParameters(testItem, groupItem) {
+    if(testItem.parameters !== null) {
+      Object.values(testItem.parameters).forEach((param) => {
+        if(param.isMandatory === false) {
+          if(this.checkIfParamExists(param.current)) {
+            groupItem.att(param.tag, param.current)
+          }
+        } else {
+          if(this.checkIfParamExists(param.current)) {
+            groupItem.att(param.tag, param.current);
+          } else if(!this.checkIfParamExists(param.current) && this.checkIfParamExists(param.default)) {
```

Comment:
I think you can skip that (!this.checkIfParamExists(param.current) as in previous conditional you are checking this.checkIfParamExists(param.current) so it should always be false later on

Reply:
Right, done.